django_project/api/api_views/v2/clustering.py

```diff
@@ -0,0 +1,86 @@
+# coding=utf-8
+__author__ = 'Anita Hapsari <anita@kartoza.com>'
+__date__ = '09/04/19'
+
+from django.contrib.gis.geos import Polygon, MultiPoint
+from rest_framework import serializers
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from sklearn.cluster import KMeans
+from pandas import DataFrame
+import numpy as np
+from localities_osm.utilities import get_all_osm_query
+from localities_osm.models.locality import LocalityOSMView
+
+
+class Clustering(APIView):
+
+    def get(self, request):
+        query = get_all_osm_query()
+
+        bbox = request.GET.get('bbox', None)
+        if bbox:
+            bbox_array = bbox.split(',')
+            bbox_array = [float(i) for i in bbox_array]
+            bbox_geom = Polygon.from_bbox(bbox_array)
+            query = query.filter(geometry__within=bbox_geom)
+
+        serializer = LocalitiesSerializer(query, many=True)
+        df = DataFrame(
+            serializer.data,
+            columns=['centroid_lat', 'centroid_lng'], index=None)
+
+        response_data = []
+        total = query.count()
+
+        if total > 0:
+            clusters = 20
```

Comment:
what is this number?

Reply:
it's the number of cluster, this one is hardcoded, but I am also thinking to find the best number of cluster through function maybe something similar to elbow method

Reply:
could you put this on setting?

Reply:
sure